staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -3515,36 +3510,36 @@ func getSGListFromAnnotation(annotatedSG string) []string {
 // Extra groups can be specified via annotation, as can extra tags for any
 // new groups. The annotation "ServiceAnnotationLoadBalancerSecurityGroups" allows for
 // setting the security groups specified.
-func (c *Cloud) buildELBSecurityGroupList(serviceName types.NamespacedName, loadBalancerName string, annotations map[string]string) ([]string, error) {
+func (c *Cloud) buildELBSecurityGroupList(serviceName types.NamespacedName, loadBalancerName string, annotations map[string]string) ([]string, bool, error) {
 	var err error
 	var securityGroupID string
+	// We do not want to make changes to a Global defined SG
+	var setupSg = false
 
 	sgList := getSGListFromAnnotation(annotations[ServiceAnnotationLoadBalancerSecurityGroups])
 
-	// The below code changes makes sure that when we have Security Groups  specified with the ServiceAnnotationLoadBalancerSecurityGroups
-	// annotation we don't create a new default Security Groups
-
 	// If no Security Groups have been specified with the ServiceAnnotationLoadBalancerSecurityGroups annotation, we add the default one.
 	if len(sgList) == 0 {
 		if c.cfg.Global.ElbSecurityGroup != "" {
-			securityGroupID = c.cfg.Global.ElbSecurityGroup
+			sgList = append(sgList, c.cfg.Global.ElbSecurityGroup)
```

Comment:
Hmm code duplication is something else for me. It would not save any line if intermediate variable is used here and the add to list is consolidated into the outer block and for me I did it to have it more readable / understandable what happens in which branch, but yes can change it back if other like the previous way more.

Reply:
If we think this improves the readability then I am good with that.

Was just thinking the codebase like kubernetes/Kubernetes if we have similar code then it would be too many duplicate coding.
Like in this change itself , the setupSg was set to false and was not updated again in the if block
```
if c.cfg.Global.ElbSecurityGroup != "" {

```
Because its duplicate but would definitely add readability that ok the value is  set to false irrespective of what has been set earlier.

Reply:
I think an append to array line is not something very different to an assignment so no additional duplicated code. But at the end I'd be fine with both ways.

For the setupSg the intention was to show: it's always as default False and should only in explicit case be set to true. Of course we could set it a 2nd time to false in `if c.cfg.Global.ElbSecurityGroup != "" {` block and another time in an else block to the `if len(sgList) == 0 {` but for me this would be too much.

Anyway at the end I'm fine to change the code style, I just want the fix :-D